### ISSUE TITLE
Grayscale image mask is transformed in a performance decreasing way with ToTensor()

### ISSUE THREAD

**Issue by Areiser** [pytorch/vision#2497]
## 🐛 Bug

Preface: This is only in the context of my usage, where I use a custom dataset for semantic segmentation. My goal was to train different models on a semantic segmentation dataset and obviously I wanted to use the torchvision transforms.
The dataset does not matter much, but the labels are interesting: The labels are grayscale in a 2d array with values from 0 to NUM_CLASSES). (I mean in form of a PIL image when I say array for the label)

I realized after training on a subset of the data, that the performance was incredibly bad and the models could not even overfit on a simple dataset.

I debugged for a while and finally realized, that the ToTensor() operation changes the arrays from (W, H) to (3, W, H) and the values become some float values because of the [0,255] to [0,1] rescaling. What I did not realize is how much of a performance impact this change has. When just using torch.tensor() to create a tensor from the array, the performance was WAY better (see chart below, the gray performance was using the torch.tensor approach). Note that in the chart, the ONLY change I made is replace the transforms (see in reproduction explanation).

Charts with MIoU and Loss, pink: HRNet with old transform, orange: DeepLabV3+ with old transform, grey: HRNet with new transform. (All pretrained on a different dataset btw)
![Screenshot from 2020-07-21 14-04-06](https://user-images.githubusercontent.com/14922864/88052834-24516d00-cb5b-11ea-8d18-164190cf329e.png)

![Screenshot from 2020-07-21 14-04-19](https://user-images.githubusercontent.com/14922864/88052842-27e4f400-cb5b-11ea-8296-5407c6024dba.png)

## To Reproduce

1. Create labelset that does not use RGB labels but uses a grayscale array with each class having a corresponding number
1. Prepare a semantic segmentation model (I used HRNet and DeepLabV3+) for training
1. Load the dataset with these transforms for the labels:  
```
label_transform = transforms.Compose([
    transforms.Resize(downsampling_size, interpolation=Image.NEAREST),
    transforms.ToTensor(),
])
```
1. Train the model, plot performance (mIoU is bad even though loss keeps decreasing)
1. Load the dataset with the following different transforms:  
```
custom_to_tensor = lambda a : torch.tensor(np.array(a))
label_transform = transforms.Compose([
    # Nearest interpolation to keep valid labels
    transforms.Resize(downsampling_size, interpolation=Image.NEAREST),
    custom_to_tensor,
])
```
1. Train again, realize the performance is way better??

## Expected behavior

ToTensor() should not lead to such a huge performance loss when using grayscale image masks :(

## Environment

Using pytorch/pytorch:1.5.1-cuda10.1-cudnn7-runtime with these pip dependencies installed:
* tensorboard==2.2.0
* matplotlib==3.2.2
* tensorboardx==2.0
* Pillow==7.2.0
* numpy==1.19.0
* python-box==5.0.1
* pytorch-ignite==0.4.0.post1 

 GPU is Nvidia Quadro P6000, only used one so far

## Additional context

I realize this might not be a bug per definition, but it still threw me for a loop. I absolutely did not expect the simple usage of ToTensor() to hinder the performance this much.

**Comment by vfdev-5** [comment]
@Areiser please, take a look at `references/segmentation` with reference training scripts for semantic segmentation. If you take a look at `ToTensor()` operation adapted for semantic segmentation:
https://github.com/pytorch/vision/blob/1aef87d01eec2c0989458387fa04baebcc86ea7b/references/segmentation/transforms.py#L78-L82

Target is not transformed with `F.to_tensor` operation. Maybe, this could be helpful and be used for your use-case.

**Comment by Areiser** [comment]
Yes as described this is exactly the approach I took. It just seemed unintuitive that I could not use the default ToTensor() for this, but I guess it is intended this way. I was not aware of the reference scripts, thanks for that. At least it is well documented both in the reference as well as the default ToTensor so I will close this issue.

**Comment by vfdev-5** [comment]
@Areiser yes I agree that probably the scope of transforms like `ToTensor()` may seem not be clear. Maybe, a note in the documentation of `ToTensor()` saying that we should not use it to transform targets to tensor can be helpful. 

PS. If you would like help to improve the project, feel free to send a PR :)
